test/interpreter/function-binom.inv.spec.ts

```diff
@@ -0,0 +1,165 @@
+    expect(engine.getCellValue(adr('A2'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.NumberCoercion))
+    expect(engine.getCellValue(adr('A3'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.NumberCoercion))
+  })
+
+  it('should work', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=BINOM.INV(10, 0.5, 0.001)',
+      '=BINOM.INV(10, 0.5, 0.01)',
+      '=BINOM.INV(10, 0.5, 0.025)',
+      '=BINOM.INV(10, 0.5, 0.05)',
+      '=BINOM.INV(10, 0.5, 0.1)',
+      '=BINOM.INV(10, 0.5, 0.2)',
+      '=BINOM.INV(10, 0.5, 0.3)',
+      '=BINOM.INV(10, 0.5, 0.4)',
+      '=BINOM.INV(10, 0.5, 0.5)',
+      '=BINOM.INV(10, 0.5, 0.6)',
+      '=BINOM.INV(10, 0.5, 0.7)',
+      '=BINOM.INV(10, 0.5, 0.8)',
+      '=BINOM.INV(10, 0.5, 0.9)',
+      '=BINOM.INV(10, 0.5, 0.95)',
+      '=BINOM.INV(10, 0.5, 0.975)',
+      '=BINOM.INV(10, 0.5, 0.99)',
+      '=BINOM.INV(10, 0.5, 0.999)'],
+    ])
+
+    expect(engine.getSheetValues(0)).toEqual([[1, 1, 2, 2, 3, 4, 4, 5, 5, 5, 6, 6, 7, 8, 8, 9, 9]])
+  })
+
+  it('should work, different p-value', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=BINOM.INV(10, 0.8, 0.001)',
+        '=BINOM.INV(10, 0.8, 0.1)',
+        '=BINOM.INV(10, 0.8, 0.2)',
+        '=BINOM.INV(10, 0.8, 0.3)',
+        '=BINOM.INV(10, 0.8, 0.4)',
+        '=BINOM.INV(10, 0.8, 0.5)',
+        '=BINOM.INV(10, 0.8, 0.6)',
+        '=BINOM.INV(10, 0.8, 0.7)',
+        '=BINOM.INV(10, 0.8, 0.8)',
+        '=BINOM.INV(10, 0.8, 0.9)',
+        '=BINOM.INV(10, 0.8, 0.999)'],
+    ])
+
+    expect(engine.getSheetValues(0)).toEqual([[4, 6, 7, 7, 8, 8, 8, 9, 9, 10, 10]])
+  })
+
+  it('should work, small number of trials', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=BINOM.INV(0, 0.8, 0.001)',
+        '=BINOM.INV(0, 0.8, 0.1)',
+        '=BINOM.INV(0, 0.8, 0.2)',
+        '=BINOM.INV(0, 0.8, 0.3)',
+        '=BINOM.INV(0, 0.8, 0.4)',
+        '=BINOM.INV(0, 0.8, 0.5)',
+        '=BINOM.INV(0, 0.8, 0.6)',
+        '=BINOM.INV(0, 0.8, 0.7)',
+        '=BINOM.INV(0, 0.8, 0.8)',
+        '=BINOM.INV(0, 0.8, 0.9)',
+        '=BINOM.INV(0, 0.8, 0.999)'],
+    ])
+
+    expect(engine.getSheetValues(0)).toEqual([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]])
+  })
+
+  it('should work, another small number of trials', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=BINOM.INV(1, 0.8, 0.001)',
+        '=BINOM.INV(1, 0.8, 0.1)',
+        '=BINOM.INV(1, 0.8, 0.2)',
+        '=BINOM.INV(1, 0.8, 0.3)',
+        '=BINOM.INV(1, 0.8, 0.4)',
+        '=BINOM.INV(1, 0.8, 0.5)',
+        '=BINOM.INV(1, 0.8, 0.6)',
+        '=BINOM.INV(1, 0.8, 0.7)',
+        '=BINOM.INV(1, 0.8, 0.8)',
+        '=BINOM.INV(1, 0.8, 0.9)',
+        '=BINOM.INV(1, 0.8, 0.999)'],
+    ])
+
+    expect(engine.getSheetValues(0)).toEqual([[0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1]])
+    //both products #1 and #2 return 0 for '=BINOM.INV(1, 0.8, 0.2)', which is incorrect
```

Comment:
We're keeping the compatibility with products #1 and #2? From what I see our result is `0` as well, but GS and XL returned `1`

Reply:
this is a border case. Jstat returns correct value. We're returning correct value. I have no idea how to "break" jstat to return incorrect value consistently (that is, so that rounding error is always in the same direction as implementation of this distribution in GS/XL). Besides, I think it is not a good direction.

Reply:
So the sentence should be:
```suggestion
    //both products #1 and #2 return 1 for '=BINOM.INV(1, 0.8, 0.2)', which is incorrect
```
Right? Then we have correctly calculated `0`, they are incorrect and the result is `1` which I see in both apps

Reply:
yes, sorry for confusion